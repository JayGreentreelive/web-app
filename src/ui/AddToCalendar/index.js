import React from 'react'
import PropTypes from 'prop-types'
import { uniqueId } from 'lodash'

import {
  Dropdown
} from 'react-bootstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faGoogle,
  faApple,
  faMicrosoft
} from '@fortawesome/free-brands-svg-icons'

import {
  googleCalLink,
  icsLink
} from './utils'

const AddToCalendar = ({
  event,
  type,
  title,
  className,
  style
}) => (
    <Dropdown>
      <Dropdown.Toggle
        id={uniqueId('add-to-calendar-')}
        variant={type}
        className={className}
        style={style}
      >
        {title}
      </Dropdown.Toggle>

      <Dropdown.Menu>
        <Dropdown.Item
          href={icsLink(event)}
        >
          <FontAwesomeIcon icon={faApple} /> Apple
        </Dropdown.Item>

        <Dropdown.Item
          href={googleCalLink(event)}
        >
          <FontAwesomeIcon icon={faGoogle} /> Google
        </Dropdown.Item>

        <Dropdown.Item
          href={icsLink(event)}
        >
          <FontAwesomeIcon icon={faMicrosoft} /> Outlook
        </Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  )

AddToCalendar.propTypes = {
  event: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    address: PropTypes.string,
    startTime: PropTypes.string.isRequired,
    endTime: PropTypes.string,
  }).isRequired,
  type: PropTypes.oneOf([
    "primary",
    "secondary",
    "success",
    "info",
    "warning",
    "danger",
    "light",
    "dark",
    "link"
  ]),
  title: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object
}

AddToCalendar.defaultProps = {
  type: 'link',
  title: 'Add to Calendar',
  className: '',
  style: {}
}

export default AddToCalendar