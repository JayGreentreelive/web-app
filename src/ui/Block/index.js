import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { camelCase, get } from 'lodash'
import VisibilitySensor from 'react-visibility-sensor'

import { htmlToReactParser } from '../../utils'
import { Layout } from '..'

import ButtonRow from '../ButtonRow'

const titleClasses = classnames(
  'font-weight-bold',
)
const subtitleClasses = classnames(
  'mt-3',
  'subtitle',
  'text-secondary'
)
const htmlClasses = classnames(
  'pb-4',
)

const Block = ({
  contentLayout,
  images,
  videos,
  imageAlt,
  imageRatio,
  subtitle,
  title,
  htmlContent,
  callToAction,
  secondaryCallToAction,
  openLinksInNewTab,
  className,
  withAnimation
}) => {
  return (
    <VisibilitySensor active={withAnimation}>
      {({ isVisible }) => {
        return (
          <div
            className={classnames(
              'col',
              'py-6',
              {
                "opacity-0": !isVisible,
                "opacity-100": isVisible,
              },
              className,
            )}
          >
            <Layout
              layout={camelCase(contentLayout)}
              className={classnames(
                "max-width-1100",
              )}
              media={get(images, '[0].sources[0].uri', null) || get(videos, '[0].sources[0].uri', null)
                ? {
                  imageUrl: get(images, '[0].sources[0].uri', ''),
                  imageAlt,
                  videoUrl: get(videos, '[0].sources[0].uri', ''),
                  ratio: imageRatio,
                  showControls: true,
                  rounded: true,
                  className: classnames({
                    "animate-slide-left-right": isVisible && contentLayout === 'right',
                    "animate-slide-right-left": isVisible && contentLayout === 'left',
                  })
                } : null}
            >
              <div
                className="max-width-800 mx-auto"
                className={classnames(
                  "max-width-800",
                  "mx-auto",
                  {
                    "animate-slide-left-right": isVisible && contentLayout === 'left',
                    "animate-slide-right-left": isVisible && contentLayout === 'right',
                  }
                )}
              >
                <h5 className={subtitleClasses}>
                  {subtitle}
                </h5>

                <h1 className={titleClasses}>
                  {title}
                </h1>

                <div className={htmlClasses}>
                  {htmlToReactParser.parse(htmlContent)}
                </div>

                <ButtonRow
                  callToAction={callToAction}
                  secondaryCallToAction={secondaryCallToAction}
                  openLinksInNewTab={openLinksInNewTab}
                />
              </div>
            </Layout>
          </div>
        )
      }}
    </VisibilitySensor>
  )
}

Block.propTypes = {
  withAnimation: PropTypes.bool,
  contentLayout: PropTypes.oneOf([
    'default',
    'inverted',
    'left',
    'right'
  ])
}

Block.defaultProps = {
  withAnimation: false,
  contentLayout: 'default'
}

export default Block