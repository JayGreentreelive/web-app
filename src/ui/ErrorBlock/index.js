import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import {
  bookImage,
  onlineIcon,
  contactIcon,
  homeIcon,
  sermonIcon
} from './images'

const ErrorLink = ({ icon, call, action }) => (
  <div className={classnames(
    'row',
    'my-3',
    'd-flex',
    'align-items-center'
  )}>
    <img
      src={icon}
      style={{
        height: 40,
        width: 'auto',
      }}
    />
    <a href={action}>
      <h4 className='mb-0 pl-2'>{call}</h4>
    </a>
  </div>
)

const ErrorBlock = ({
}) => {

  return (
    <div
      className={classnames(
        'container',
        'd-flex',
        'flex-column',
        'align-items-center',
        'my-6'
      )}
    >
      <img
        src={bookImage}
        style={{
          maxWidth: 700,
        }}
      />
      <h1
        className={classnames(
          'mt-4'
        )}
      >
        Page Not Found
      </h1>
      <h4
        className={classnames(
          'font-weight-normal',
          'my-2'
        )}
      >
        The page you’re looking for doesn’t exist, or is unavailable. Here’s some helpful links instead:
      </h4>
      <div className={classnames(
        'row',
        'my-4',
        'w-75'
      )}>
        <div className='col-6'>
          <ErrorLink
            icon={onlineIcon}
            call='Watch Church Online'
            action='https://christfellowshipchurch.online.church/'
          />
          <ErrorLink
            icon={homeIcon}
            call='Go to Home Page'
            action='/'
          />
        </div>
        <div className='col-6'>
            <ErrorLink
              icon={sermonIcon}
              call='Watch the Latest Sermon'
              action='/browse'
            />
            <ErrorLink
              icon={contactIcon}
              call='Contact Us'
              action={`mailto:hello@christfellowship.church`} 
            />
        </div>
      </div>
    </div>
  )
}

ErrorBlock.propTypes = {

}

ErrorBlock.defaultProps = {

}

export default ErrorBlock