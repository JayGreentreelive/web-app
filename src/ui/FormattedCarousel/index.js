import FormattedCarousel from './FormattedCarousel'

export default FormattedCarousel