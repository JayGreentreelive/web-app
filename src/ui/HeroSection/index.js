import React, { useState } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import VisibilitySensor from 'react-visibility-sensor'
import { get } from 'lodash'

import {
  Block,
  Button,
  Media
} from '../../ui'
import {
  htmlToReactParser,
  redirectTo
} from '../../utils'

import SwoopImg from '../../images/cyan_hole_punch.svg'

const Swoop = () => {
  return (
    <img
      src={SwoopImg}
      style={{
        zIndex: 0,
        width: '105%'
      }}
      className={classnames(
        'absolute-center',
        'h-100',
        'large-background-swoop'
      )}
    />
  )
}

const HeroSection = ({
  title,
  htmlContent,
  callToAction,
  secondaryCallToAction,
  image,
  video
}) => {
  return (
    <VisibilitySensor
      active
      partialVisibility
      minTopValue={0}
    >
      {({ isVisible }) => {
        return (
          <Media
            ratio="16by9"
            overlay='black'
            videoUrl={get(video, 'uri', null)}
            imageUrl={get(image, 'uri', '')}
            imageAlt={get(image, 'alt', '')}
            fill="screen"
            className={classnames(
              'vw-100',
              'vh-100',
              "d-flex",
              "justify-content-center",
              "align-items-center",
            )}
          >
            <Swoop />

            <div
              className="w-100 max-width-1100"
              style={{ zIndex: 1000 }}
            >
              <div
                className={classnames(
                  'hero',
                  "max-width-600",
                  'p-3',
                  'pt-5',
                  {
                    'opacity-0': !isVisible,
                    'opacity-100': isVisible,
                    'animate-slide-left-right': isVisible,
                  }
                )}
              >
                <h1 className="text-white">
                  {title}
                </h1>

                <p className="my-4 text-white">
                  {htmlToReactParser.parse(htmlContent)}
                </p>

                {callToAction &&
                  <div>
                    <Button
                      className="my-2 min-width-250"
                      type="white"
                      title={get(callToAction, 'call', '')}
                      onClick={() => redirectTo(get(callToAction, 'action', ''))}
                    />
                  </div>}

                {secondaryCallToAction &&
                  <div>
                    <Button
                      className="my-2 pl-0 text-white min-width-250"
                      type="link"
                      title={get(secondaryCallToAction, 'call', '')}
                      onClick={() => redirectTo(get(secondaryCallToAction, 'action', ''))}
                    />
                  </div>}
              </div>
            </div>
          </Media>
        )
      }}
    </VisibilitySensor>
  )
}

HeroSection.propTypes = {
  title: PropTypes.string.isRequired,
  htmlContent: PropTypes.string,
  callToAction: PropTypes.shape({
    call: PropTypes.string.isRequired,
    action: PropTypes.string.isRequired,
  }),
  secondaryCallToAction: PropTypes.shape({
    call: PropTypes.string.isRequired,
    action: PropTypes.string.isRequired,
  }),
  image: PropTypes.shape({
    uri: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
  }),
  video: PropTypes.shape({
    uri: PropTypes.string.isRequired,
  })
}

HeroSection.defaultProps = {
  htmlContent: '',
  callToAction: null,
  secondaryCallToAction: null,
  image: null,
  video: null,
}

export default HeroSection