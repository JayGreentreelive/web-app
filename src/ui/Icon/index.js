import React from 'react'
import PropTypes from 'prop-types'
import icons from '../../icons.json'

const Icon = ({ type, size, fill }) => {
  
  const iconShape = icons[type]
  
  return (
  <svg xmlns="http://www.w3.org/2000/svg" width={size} height={size} viewBox="0 0 32 32" aria-labelledby="title">
    <path 
      d={iconShape}
      fill={fill}
    />
  </svg>
)}

export default Icon

Icon.propTypes = {
  size: PropTypes.number,
  fill: PropTypes.string,
  icon: PropTypes.string
};

Icon.defaultProps = {
  size: 32,
  type: 'home'
}