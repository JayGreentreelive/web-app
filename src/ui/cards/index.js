export { default as Card } from './EmptyCard'
export { default as FloatingCard } from './FloatingCard'
export { default as ContentCard } from './ContentCard'