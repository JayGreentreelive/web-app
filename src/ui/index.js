
import Accordion from './Accordion'
import BackgroundContentBlock from './BackgroundContentBlock'
import Block from './Block'
import BlockLayout from './BlockLayout'
import Button from './Button'
import ButtonRow from './ButtonRow'
import Carousel from './Carousel'
import ContentContainer from './ContentContainer'
import FloatingCard from './FloatingCard'
import GroupBlock from './GroupBlock'
import Layout from './Layout'
import Loader from './Loader'
import Media from './Media'
// import Row from './Row'


export { 
    Accordion, 
    BackgroundContentBlock, 
    Block, 
    BlockLayout, 
    Button, 
    ButtonRow, 
    Carousel,
    ContentContainer,
    FloatingCard,
    GroupBlock,
    Layout,
    Loader,
    Media
}

export {
    TextInput, Dropdown
} from './inputs'