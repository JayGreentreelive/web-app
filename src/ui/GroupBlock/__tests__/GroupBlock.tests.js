import React from 'react'
import renderer from "react-test-renderer"
import { Row } from '../../../ui'
import ContentBlock from '../../ContentBlock'
import FormattedCarousel from '../../FormattedCarousel'


describe("GroupBlock Renderer", () => {
  it("Displays a Row GroupBlock", () => {
    // const groupLayout = 'row'
    // expect(GroupBlock(groupLayout)).toBe('row')
    const tree = renderer.create(
      <Row>
        <ContentBlock/>
      </Row>  
    )
    expect(tree).toMatchSnapshot()
  })

  it("Displays a Carousel GroupBlock", () => {
    // const groupLayout = 'carousel'
    // expect(GroupBlock(groupLayout)).toBe('carousel')

    const children = [{
            images: '',
            imageAlt: '',
    }]

    const tree = renderer.create(
      <FormattedCarousel children={children} />
    )
    expect(tree).toMatchSnapshot()
  })
})
