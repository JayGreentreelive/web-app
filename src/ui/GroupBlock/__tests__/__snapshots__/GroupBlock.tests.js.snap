// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupBlock Renderer Displays a Accordion GroupBlock 1`] = `
<div>
  <div
    class="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light"
  >
    <div
      class="fill d-flex justify-content-center align-items-center"
    >
      <div
        class="spinner-border text-primary"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`GroupBlock Renderer Displays a Carousel GroupBlock 1`] = `
<div>
  <div
    class="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light"
  >
    <div
      class="fill d-flex justify-content-center align-items-center"
    >
      <div
        class="spinner-border text-primary"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`GroupBlock Renderer Displays a Row GroupBlock 1`] = `
<div>
  <div
    class="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light"
  >
    <div
      class="fill d-flex justify-content-center align-items-center"
    >
      <div
        class="spinner-border text-primary"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
